subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/tasks/ValidateTaskProperties.java

```diff
@@ -307,26 +316,52 @@ public boolean getFailOnWarning() {
 
     /**
      * Returns the output file to store the report in.
+     *
+     * @since 4.5
      */
-    @Optional @OutputFile
-    public File getOutputFile() {
-        return outputFile == null ? null : getProject().file(outputFile);
+    @Optional
+    @OutputFile
+    public RegularFileProperty getOutputFile() {
+        return outputFile;
     }
 
     /**
      * Sets the output file to store the report in.
      *
      * @since 4.0
+     *
+     * @deprecated Use {@link #getOutputFile()} instead.
      */
+    @Deprecated
     public void setOutputFile(File outputFile) {
```

Comment:
We automatically create a setter for the `getOutputFile` which takes a `File` and is accessible from Groovy, so you could write:
```
validateTaskProperties.outputFile = new File('some')
```

If we deprecate this setter we cannot use the decoration from Groovy, can we?
OTOH, removing the setter may break statically compiled plugins using this task.
Ideas?

Reply:
Yes, and removing the setter would break Java compatibility, too. Not that I think either is used.

Reply:
I'd just let this method expire in 5.0.

Reply:
OK, together with the other setter I'm convinced we should just remove these guys, since `@Incubating`.